src/receive_imf.rs

```diff
@@ -1639,88 +1644,125 @@ async fn apply_group_changes(
         return Ok(None);
     }
 
-    let mut recreate_member_list = false;
     let mut send_event_chat_modified = false;
-
+    let mut removed_id = None;
     let mut better_msg = None;
-    let removed_id;
+    let mut apply_group_changes = false;
+
+    let mut recreate_member_list = if mime_parser
+        .get_header(HeaderDef::ChatGroupMemberRemoved)
+        .is_some()
+        || mime_parser
+            .get_header(HeaderDef::ChatGroupMemberAdded)
+            .is_some()
+    {
+        let self_added =
+            if let Some(added_addr) = mime_parser.get_header(HeaderDef::ChatGroupMemberAdded) {
+                if let Ok(self_addr) = context.get_primary_self_addr().await {
+                    self_addr == *added_addr
+                } else {
+                    false
+                }
+            } else {
+                false
+            };
+
+        if !chat::is_contact_in_chat(context, chat_id, ContactId::SELF).await? && !self_added {
+            false
+        } else {
+            apply_group_changes = chat_id
+                .update_timestamp(context, Param::MemberListTimestamp, sent_timestamp)
+                .await?;
+
+            match mime_parser.get_header(HeaderDef::InReplyTo) {
```

Comment:
Why do we check this only on a receipt of MemberAdded/Removed messages? Looks like it should be checked always (and have a priority over MemberAdded/Removed checks), otherwise we can miss member changes